protocols/gossipsub/src/behaviour.rs

```diff
@@ -3260,7 +3260,15 @@ where
                 let mut ihave_msgs = vec![];
                 let mut graft_msgs = vec![];
                 let mut prune_msgs = vec![];
-                for control_msg in rpc.control_msgs {
+                for (count, control_msg) in rpc.control_msgs.into_iter().enumerate() {
+                    // Only process the amount of messages the configuration allows.
+                    if self.config.max_messages_per_rpc().is_some()
+                        && Some(count) >= self.config.max_messages_per_rpc()
```

Comment:
we could use `Option::is_some_and` to perform the check. Eg

```suggestion
                        && self.config.max_messages_per_rpc().is_some_and(|max_msg| count >= max_msg)
```

Probably could get away with just that statement check

Reply:
thanks Darius, looks better!